pybop/models/lithium_ion/basic_SPMe.py

```diff
@@ -0,0 +1,619 @@
+        zeta_p = Parameter("Positive electrode relative porosity")
+
+        tau_e_n = Parameter("Negative electrode electrolyte diffusion time scale [s]")
+        tau_e_sep = Parameter("Separator electrolyte diffusion time scale [s]")
+        tau_e_p = Parameter("Positive electrode electrolyte diffusion time scale [s]")
+
+        ######################
+        # Input current (positive on discharge)
+        ######################
+        I = self.param.current_with_time
+
+        ######################
+        # State of Charge
+        ######################
+        # The `rhs` dictionary contains differential equations, with the key being the
+        # variable in the d/dt
+        self.rhs[Q] = I / 3600
+        self.rhs[Qt] = abs(I) / 3600
+        # Initial conditions must be provided for the ODEs
+        self.initial_conditions[Q] = Scalar(0)
+        self.initial_conditions[Qt] = Scalar(0)
+
+        ######################
+        # Overpotentials
+        ######################
+        U_n = self.U(sto_n_surf, "negative")
+        U_p = self.U(sto_p_surf, "positive")
+        if include_double_layer:
+            eta_n = (v_s_n - U_n) + (2 * RT_F * (1 - t_plus)) * (
+                pybamm.x_average(pybamm.log(sto_e_n)) - pybamm.log(sto_e_n)
+            )
+            eta_p = (v_s_p - U_p) + (2 * RT_F * (1 - t_plus)) * (
+                pybamm.x_average(pybamm.log(sto_e_p)) - pybamm.log(sto_e_p)
+            )
+        eta_e = (2 * RT_F * (1 - t_plus)) * (
+            pybamm.x_average(pybamm.log(sto_e_p))
+            - pybamm.x_average(pybamm.log(sto_e_n))
+        )
+
+        ######################
+        # Exchange current
+        ######################
+        # Primary broadcasts are used to broadcast scalar quantities across a domain
+        # into a vector of the right shape, for multiplying with other vectors
+
+        alpha = 0.5  # cathodic transfer coefficient
+        j0_n = (
+            sto_n_surf**alpha * (sto_e_n * (1 - sto_n_surf)) ** (1 - alpha) / tau_ct_n
+        )
+        j0_p = (
+            sto_p_surf**alpha * (sto_e_p * (1 - sto_p_surf)) ** (1 - alpha) / tau_ct_p
+        )
+        if include_double_layer:
+            j_n = j0_n * (
+                pybamm.exp((1 - alpha) * eta_n / RT_F)
+                - pybamm.exp(-alpha * eta_n / RT_F)
+            )
+            j_p = j0_p * (
+                pybamm.exp((1 - alpha) * eta_p / RT_F)
+                - pybamm.exp(-alpha * eta_p / RT_F)
+            )
+        else:
+            # Assuming alpha = 0.5
+            j_n = PrimaryBroadcast(I / (3 * Q_th_n), "negative electrode")
+            j_p = PrimaryBroadcast(-I / (3 * Q_th_p), "positive electrode")
+            eta_n = 2 * RT_F * pybamm.arcsinh(j_n / (2 * j0_n))
+            eta_p = 2 * RT_F * pybamm.arcsinh(j_p / (2 * j0_p))
+            v_s_n = pybamm.x_average(eta_n + U_n)
+            v_s_p = pybamm.x_average(eta_p + U_p)
```

Comment:
And those maybe in the next if statement, so they all include the if/else for the same variables